kube-runtime/src/watcher.rs

```diff
@@ -217,6 +234,21 @@ pub struct Config {
     /// Configures re-list for performance vs. consistency.
     pub list_semantic: ListSemantic,
```

Comment:
should caveat that options such as this do not always have an effect anymore:

```suggestion
    /// Configures re-list for performance vs. consistency.
    ///
    /// NB: This option only has an effect for [`WatcherMode::ListWatch`].
    pub list_semantic: ListSemantic,
```

we probably should also add this to `::page_size` plus the builders `::any_semantic()`, `::list_semantic()`, `::page_size()`.